### DISCUSSION TITLE
Solana Test Validator and Metaplex programs

### DISCUSSION THREAD

**Discussion by katewayo** [metaplex-foundation/js#69] [Q&A] [answered]
Hi! As part of my development environment, I need to run the Solana Test Validator with all the relevant Metaplex programs loaded such that the js-next operations, e.g. createNft, can succeed. Unfortunately, I haven't been able to find a comprehensive list of program ids to clone into the test validator (or a cookbook for this kind of setup). Without that, I keep getting the following from calling createNft:

SendTransactionError: failed to send transaction: Transaction simulation failed: Attempt to load a program that does not exist

Note that I can't use devnet because I'm actually writing the NFT metadata generation code (that responds to the URI request) and that's coming from my localhost server. Hopefully that all makes sense.

Is there a simple cookbook for setting up a test validator environment with all the programs that need to be called from js-next? I'd prefer to clone them from mainnet via --clone on validator startup vs. building and loading them locally. That said, I can do the latter if I had a clear set of instructions. Thanks!

**Comment by lorisleiva** [comment, marked as answer]
Hi there 👋  That's a good question. We need to work on making this more easily accessible to developers and write some content to get them started. In the meantime, I'd recommend using "Amman" which is a wrapper around test validators that makes it easier for us to test the JS SDK. For that you need to clone the `js-next` repo and run `yarn && yarn amman:start` inside it. This will start a local test validator with the TokenMetadata program loaded into it. Let me know if you have any questions/issues. 🤞

**Reply by katewayo** [reply]
That did it! Thank you!!!!

**Reply by jakerumbles** [reply]
I'm not finding this js-next repository in your organization. Was it deprecated and the `amman` stuff moved somewhere else?

**Reply by jakerumbles** [reply]
Ok I see there is a `js` repo, but it's been deprecated. Did the `amman` stuff move to `umi` or a different repo?
